v0.5/fastn-p1/src/parser/scanner.rs

```diff
@@ -134,18 +136,28 @@ impl Scanner {
 
     // returns the span from current position to the end of token
     pub fn token(&mut self, t: &'static str) -> Option<fastn_p1::Span> {
-        let mut count = 0;
-        for char in t.chars() {
+        // Get the length of the token to match
+        let token_len = t.chars().count();
```

Comment:
Keep my `mut count` solution. You are going through `t.chars()` twice. No need to do that.

Reply:
But `if char != self.tokens[self.index + count] {` can overflow which `if self.index + token_len > self.size` took care of

Reply:
Put an `&& self.index + count < self.size` there. 

Ideally we should look at generated assembly to see which one is shorter. But my point is let's try to really worry about performance this time.